app/screens/user_profile/user_info.tsx

```diff
@@ -21,36 +21,50 @@ type Props = {
     showPosition: boolean;
     user: UserModel;
     enableCustomAttributes?: boolean;
+    customAttributesSet?: CustomAttributeSet;
 }
 
 const emptyList: CustomAttribute[] = []; /** avoid re-renders **/
 
-const UserInfo = ({localTime, showCustomStatus, showLocalTime, showNickname, showPosition, user, enableCustomAttributes}: Props) => {
+const UserInfo = ({
+    localTime,
+    showCustomStatus,
+    showLocalTime,
+    showNickname,
+    showPosition,
+    user,
+    enableCustomAttributes,
+    customAttributesSet,
+}: Props) => {
     const customStatus = getUserCustomStatus(user);
     const serverUrl = useServerUrl();
     const [customAttributes, setCustomAttributes] = useState<CustomAttribute[]>(emptyList);
-
     const lastRequest = useRef(0);
 
+    // Initial load from database and server if customAttributesSet is not provided
     useEffect(() => {
         if (enableCustomAttributes) {
-            const fetchData = async () => {
+            // If customAttributesSet is provided by the parent, use it
+            if (customAttributesSet && Object.keys(customAttributesSet).length > 0) {
+                setCustomAttributes(Object.values(customAttributesSet).sort(sortCustomProfileAttributes));
+                return;
+            }
+
+            const fetchFromServer = async () => {
```

Comment:
Do we only fetch attributes here? If so, we may have a problem with reconciliation.

Imagine I have attributes loaded in the database. I close the app, and those attributes get deleted. Since the app was closed, I don't receive the websocket messages.

Now I open the app, open the profile of this user. This logic will say "I have information in the database, so do not fetch again". So I never realize some attributes has been deleted (or even updated!).

Is my supposition right, or am I missing something?

Reply:
no, you are right, and I had to think about this when I was creating the feature. We aren't using the info anywhere but when clicking on the user profile or when editing our own profile. While we can fetch on start/wakeup/etc. the truth is that this info is only relevant when looking at the profile. If this was critical in anyway, I'd definitely go for syncing every time we connect to the server, but I think that would only create overhead as we would also have to fetch from every possible user we share a channel with.
So my decision (until someone says I'm wrong and we should do it some other way) is to only fetch when it is really relevant. If at that moment we are not connected, tough luck, but I think it is a good compromise.

Reply:
I agree with that approach. But that being said, what this code is doing is to fetch only if we don't have local data. So that synchronization is never done after you have the first value.

I would say we just have to remove the return in line 50.

I have mixed feelings on this approach for a couple of reasons:
- You never know how old is the data when looking at it offline
- You still have to make a fetch every time you look at the information

There are of course ways to address both problems, but I agree with you it is so far a good compromise. That being said, I wonder if we should confirm with the stakeholders for this feature.

Reply:
> I would say we just have to remove the return in line 50.

Wow, that skipped, but you are right. Removed.

maybe we can create a ticket for adding some sort of badge when looking at info that it is cached, do we have a similar approach somewhere else?
cc/ @asaadmahmood 

Regarding the fetch, I don't have data for how common it is to look at your/others' profiles, but if we have that, then we can make better decisions. Being an unknown, I'm ok with the current compromise.